lib/index.js

```diff
@@ -48,11 +48,16 @@ function S3(uri, callback) {
     // If a parsed uri of the form s3://[bucket]/[path template] is passed,
     // assume that tiles are implied (no grids), and generate a data
     // object containing an https://[bucket].s3.amazonaws.com/[path template]
-    // tiles key.
+    // tiles key. If region flag is waved and there exists an s3endpoint in
+    // the regionPartitions index, that will be used for the hostname.
 
     if (uri.protocol === 's3:' && uri.host) {
+        var host = uri.host + '.s3.amazonaws.com';
+        if (query && query.region) {
+            host = regionParitions(query.region, 's3endpoint') || host;
```

Comment:
Do we really need to add the complexity of an additional `regionPartitions` function? Also just noticed that there's a typo in the code here.

Instead we can just deal with the only outlier - the cn-north-1 situation 

```js
if (query && query.region === 'cn-north-1') {
  host = 's3.cn-north-1.amazonaws.com.cn';
}
```